### ISSUE TITLE
Gradient clipping by norm has different semantics in tf.keras.optimizers against keras.optimizers

### ISSUE THREAD

**Issue by chenchc** [tensorflow/tensorflow#29108]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 1.13.1
- Python version: 3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: CUDA 10.0 + cuDNN 7.4
- GPU model and memory: NVIDIA Tesla V100 32GB

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

The gradient clipping mechanism is implemented in the abstract class, tf.keras.optimizers.Optimizer (keras.optimizers.Optimizer) so that every optimizer inherited from the class supports it.
We find different semantics of the implementations between tf.keras and keras.io.

In tf.keras, the gradient norm is calculated and clipped **per gradient tensor**.

https://github.com/tensorflow/tensorflow/blob/bb8cf258bc87f68612a5e032d5b4def0d3c52566/tensorflow/python/keras/optimizers.py#L98-L99

But in keras.io, the gradient norm is calculated globally **across all gradient tensors**.

```python
        if hasattr(self, 'clipnorm') and self.clipnorm > 0:
            norm = K.sqrt(sum([K.sum(K.square(g)) for g in grads]))
            grads = [clip_norm(g, self.clipnorm, norm) for g in grads]
```
Code link: https://github.com/keras-team/keras/blob/eab1b5bcdf105746ede02d2eb8a5cb3ca359b1b5/keras/optimizers.py#L96-L98

IMO, tf.keras should adopt the latter method since
(1) The paper introducing gradient clipping (https://arxiv.org/abs/1211.5063) suggests the latter one, and
(2) tf.keras should be compliant with keras.io.

**Comment by tanzhenyu** [comment]
I already provided comment in the PR. Closing this.

**Comment by chenchc** [comment]
Hi @tanzhenyu , thanks for your reply. 
But where can I find the comment you left?

**Comment by tanzhenyu** [comment]
@chenchc in the PR, go to files changes tab (or the regular page) you should be able to see it

**Comment by chenchc** [comment]
@tanzhenyu I cannot see the comments in both tabs. Maybe they are invisible to my account?

<img width="1015" alt="Screen Shot 2019-06-26 at 12 37 12" src="https://user-images.githubusercontent.com/6285919/60151408-33a69f80-980f-11e9-8cf6-360f5933b1f0.png">

**Comment by tanzhenyu** [comment]
@chenchc My major comment is that I don't see where in this paper describing it should use global clip instead of local clip.
